lib/chef/knife/bmcs_server_delete.rb

```diff
@@ -29,16 +29,38 @@ class BmcsServerDelete < Knife
              long: '--wait SECONDS',
              description: 'Wait for the instance to be terminated. 0=infinite'
 
+      option :prune,
+             long: '--prune',
+             description: 'Also remove node from Chef server.'
+
+      option :chef_node_name,
+             long: '--chef-node-name NAME',
+             description: 'Name of the Chef node being removed. Defaults to the instance display name.'
+
       def run
         $stdout.sync = true
         validate_required_params(%i[instance_id], config)
         wait_for = validate_wait
+        if config[:chef_node_name] && !config[:prune]
+          error_and_exit('--chef-node-name requires --prune argument')
+        end
 
-        confirm_deletion
-
-        check_can_access_instance(config[:instance_id])
+        response = check_can_access_instance(config[:instance_id])
+
+        ui.msg "Instance name: #{response.data.display_name}"
+        deletion_prompt = 'Delete server? (y/n)'
+        chef_node = nil
+        if config[:prune]
+          deletion_prompt = 'Delete server and chef node? (y/n)'
+          node_name = response.data.display_name
+          node_name = config[:chef_node_name] if config[:chef_node_name]
+          chef_node = get_chef_node(node_name)
```

Comment:
Here is the output in that case:
[vagrant@localhost knife-bmcs]$ knife bmcs server delete ... --prune --chef-node-name doesnotexist
Instance name: johntest
ERROR: The object you are looking for could not be found
Response: node 'doesnotexist' not found